testplan/web_ui/testing/src/Report/BatchReportBeta/components/AutoSelectRedirect.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { Redirect } from 'react-router';
+import connect from 'react-redux/es/connect/connect';
+import _isObjectLike from 'lodash/isObjectLike';
+import _isString from 'lodash/isString';
+import uriComponentCodec from '../../../Common/uriComponentCodec';
+import { setSelectedEntry } from '../state/uiActions';
+import { setHashComponentAlias } from '../state/uiActions';
+import { mkGetUIDoAutoSelect } from '../state/uiSelectors';
+import { mkGetUIHashAliasToComponent } from '../state/uiSelectors';
+import { BOTTOMMOST_ENTRY_CATEGORY } from '../../../Common/defaults';
+
+const connector = connect(
+  () => {
+    const getDoAutoSelect = mkGetUIDoAutoSelect();
+    const getAliasToComponent = mkGetUIHashAliasToComponent();
+    return state => ({
+      doAutoSelect: getDoAutoSelect(state),
+      aliasToComponent: getAliasToComponent(state),
+    });
+  },
+  { setSelectedEntry, setHashComponentAlias },
+  (stateProps, dispatchProps, ownProps) => {
+    const { doAutoSelect, aliasToComponent } = stateProps;
+    const { setSelectedEntry, setHashComponentAlias } = dispatchProps;
+    const { location, entry } = ownProps;
+    return {
+      getRedirectToEntry: () => {
+        const to = { ...location, pathname: (
+          `${location.pathname.replace(/\/+$/, '').trim()}/${entry.name || ''}`
+        )};
+        let currEntry = entry, nextEntry, nextAlias;
+        while(
+          doAutoSelect
+          && _isObjectLike(currEntry)
+          && currEntry.category !== BOTTOMMOST_ENTRY_CATEGORY
+          && Array.isArray(currEntry.entries)
+          && currEntry.entries.length === 1
+          && _isObjectLike(nextEntry = currEntry.entries[0] || {})
```

Comment:
why are we always going into the first entry?

Reply:
this is the autoselect logic that was already present but this is checking if there's only one entry and if so it selects the 1st = last = only entry